pkg/apis/aerospike/v1alpha1/aerospikecluster_types.go

```diff
@@ -366,21 +434,32 @@ func (v *AerospikeStorageSpec) SetDefaults() {
 		// Use storage spec values as defaults for the volumes.
 		if v.Volumes[i].VolumeMode == AerospikeVolumeModeBlock {
 			v.Volumes[i].AerospikePersistentVolumePolicySpec.SetDefaults(&v.BlockVolumePolicy)
-		} else {
+		} else if v.Volumes[i].VolumeMode == AerospikeVolumeModeFilesystem {
 			v.Volumes[i].AerospikePersistentVolumePolicySpec.SetDefaults(&v.FileSystemVolumePolicy)
 		}
 	}
 }
 
 // GetStorageList gives blockStorageDeviceList and fileStorageList
 func (v *AerospikeStorageSpec) GetStorageList() (blockStorageDeviceList []string, fileStorageList []string, err error) {
-	storagePaths := map[string]int{}
+	storagePaths := map[string]int{
```

Comment:
If this path is also given by user then It would be better to give error saying that this path is reserved.

Reply:
Yes will change.